test/unified/runner.js

```diff
@@ -0,0 +1,366 @@
+'use strict';
+const expect = require('chai').expect;
+const path = require('path');
+const semver = require('semver');
+const MongoClient = require('../../src/mongo_client').MongoClient;
+
+class Version {
+  static clean(version) {
+    return semver.coerce(version).version;
+  }
+  static check(supportedSchemaVersions, schemaVersion) {
+    let satisfies = false;
+    supportedSchemaVersions.map(Version.clean).forEach(v => {
+      if (satisfies) return;
+      const result = semver.satisfies(v, `^${Version.clean(schemaVersion)}`);
+      if (result) satisfies = true;
+    });
+    return satisfies;
+  }
+  static validate(supportedSchemaVersions, spec) {
+    const check = Version.check(supportedSchemaVersions, spec.schemaVersion);
+    if (!check) throw new Error(`invalid schemaVersion: ${spec.description}`);
+  }
+}
+
+class Match {
```

Comment:
Though it does appear that we've lost the test suite for it, we have a very stable [custom chai matcher](https://github.com/mongodb/node-mongodb-native/blob/master/test/functional/spec-runner/matcher.js) already written that follows today's matching rules. I would prefer that we improve that code, and thoroughly test it, than introduce a new one.

Reply:
This is great and I'm totally down to use it where it fits, but we still have to match object properties and I still believe there is some novelty here.